src/atomate2/openmm/jobs/core.py

```diff
@@ -0,0 +1,328 @@
+    ):
+        """
+        Create a flow that runs in the NPT ensemble.
+
+        Parameters
+        ----------
+        input_set : OpenMMSet
+            A pymatgen structure object.
+        platform : Optional[Union[str, openmm.openmm.Platform]]
+             platform: the OpenMM platform passed to the Simulation.
+        platform_properties : Optional[Dict[str, str]]
+            properties of the OpenMM platform that is passed to the simulation.
+
+        Returns
+        -------
+        Job
+            A OpenMM job containing one npt run.
+        """
+        platform = Platform.getPlatformByName(platform)
+        sim = input_set.get_simulation(
+            platform=platform,
+            platformProperties=platform_properties
+        )
+        sim.minimizeEnergy()
+        new_state_file_path = os.path.join(output_dir, "state_xml")
+        sim.saveState(new_state_file_path)
+        return input_set
+
+
+@dataclass
+class NPTMaker(BaseOpenmmMaker):
+    steps: int = 1000000
+    name: str = "npt simulation"
+    temperature: float = 298
+    pressure: float = 1
+    frequency: int = 10
+
+    @job
+    def make(
+            self,
+            input_set: OpenMMSet,
+            output_dir: Union[str, Path],
+            platform: Optional[Union[str, Platform]] = "CPU",
+            platform_properties: Optional[Dict[str, str]] = None,
+    ):
+        """
+        Equilibrate the pressure of a simulation in the NPT ensemble.
+
+        Adds and then removes a openmm.MonteCarlo Barostat to shift the system
+        into the NPT ensemble.
+
+        Parameters
+        ----------
+        input_set : OpenMMSet
+            A pymatgen structure object.
+        platform : Optional[Union[str, openmm.openmm.Platform]]
+             platform: the OpenMM platform passed to the Simulation.
+        platform_properties : Optional[Dict[str, str]]
+            properties of the OpenMM platform that is passed to the simulation.
+
+        Returns
+        -------
+        Job
+            A OpenMM job containing one npt run.
+        """
+
+        platform = Platform.getPlatformByName(platform)
+        sim = input_set.get_simulation(
+            platform=platform,
+            platformProperties=platform_properties,
+        )
+
+        context = sim.context
+        system = context.getSystem()
+        assert (
+            system.usesPeriodicBoundaryConditions()
+        ), "system must use periodic boundary conditions for pressure equilibration."
+        barostat_force_index = system.addForce(
+            MonteCarloBarostat(self.pressure * atmosphere, self.temperature * kelvin, 10)
+        )
+        context.reinitialize(preserveState=True)
+        sim.step(self.steps)
+        system.removeForce(barostat_force_index)
+        context.reinitialize(preserveState=True)
+        new_state_file_path = os.path.join(output_dir, "state_xml")
+        sim.saveState(new_state_file_path)
```

Comment:
Let's try replacing this with the following. This way we won't save the parameters that are causing issues.

```python
from pymatgen.io.openmm.inputs import StateInput  # belongs at top

state = StateInput(context.getState(
    getPositions=True,
    getVelocities=True,
    getForces=True,
    getEnergy=True,
    enforcePeriodicBox=True,
))
StateInput.to_file(state, os.path.join(output_dir, "state_xml"))
```

We should maybe leave out getForces and getEnergy arguments too. I'll have to think on that.

Reply:
I've integrated this approach into the code and it appears to be working.